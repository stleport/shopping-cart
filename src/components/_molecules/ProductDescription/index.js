import React from 'react';
import PropTypes from 'prop-types';

const ProductDescription = ({ title, description, quantity, price, view }) => {
  const blockName = view === 'card' ? 'Card' : 'Item';
  return (
    <React.Fragment>
      <div className={`st-${blockName}__description`}>
        <h2 className={`st-${blockName}__title st-${blockName}__title--${view === 'card' ? 'small' : 'big'}`} >{title}</h2>
        {view === "cart" &&
        <div className={`st-${blockName}__content`}>
          <p className={`st-${blockName}__description-text`}>
            {description}
          </p>
          <div className={`st-${blockName}__pricing`}>
            Quantit√© :&#160;
            {quantity}
          </div>
        </div>  
        }    
      </div>
    </React.Fragment>
  )
};

ProductDescription.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  quantity: PropTypes.number.isRequired
};

export default ProductDescription;
