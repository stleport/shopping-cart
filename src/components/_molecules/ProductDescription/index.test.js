import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ProductDescription from '.';

function ProductDescriptionSetup() {
  const props = {
    title: 'Title',
    description: 'Description',
    quantity: 2
  };

  const ProductDescriptionJSX = (
    <ProductDescription {...props} />
  );

  const ProductDescriptionComponent = shallow(ProductDescriptionJSX);
  return {
    ProductDescriptionComponent,
    ProductDescriptionJSX,
    props
  };
}

describe('ProductDescription', () => {
  it('renders correctly ProductDescription component', () => {
    const { ProductDescriptionJSX } = ProductDescriptionSetup();
    const ProductDescriptionComponent = renderer.create(ProductDescriptionJSX).toJSON();
    expect(ProductDescriptionComponent).toMatchSnapshot();
  });
});
