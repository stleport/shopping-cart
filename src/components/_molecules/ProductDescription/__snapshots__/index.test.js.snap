// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDescription renders correctly ProductDescription component 1`] = `
Array [
  <h2>
    Title
  </h2>,
  <p
    className="description"
  >
    Description
  </p>,
  <div
    className="st-Item__pricing"
  >
    Quantité : 
    2
  </div>,
]
`;
